src/main/java/uk/gov/hmcts/reform/preapi/services/RecordingService.java

```diff
@@ -133,6 +144,11 @@ public UpsertResult upsert(CreateRecordingDTO createRecordingDTO) {
         recordingRepository.save(recordingEntity);
 
         var isUpdate = recording.isPresent();
+
+        if (!isUpdate) {
+            onRecordingCreated(recordingEntity);
+        }
```

Comment:
You could use the `RecordingListener` class for this instead of having this in the service